### ISSUE TITLE
Document how to proceed if resource not handled automatically

### ISSUE THREAD

**Issue by sudeephb** [canonical/hardware-observer-operator#75]
If, for whatever reason, the required resource installation (eg. ipmi, ssacli, etc) doesn't happen automatically, the user has to install them manually. We need to document clearly for each resource on how to do that.

**Comment by dashmage** [comment]
Will add this as part of the documentation rewrite for hardware-observer that I'm currently working on.

**Comment by dashmage** [comment]
Is this documentation required for when a resource that's supposed to be automatically installed after detection (eg: IPMI, SSACLI) is not working? Where the resource is being detected correctly but not being installed and the user needs to do that?

In case of the other resources, the user is manually uploading them anyway and the charm doesn't need to install anything. I've raised a separate issue (#79) for the situation where the user needs to manually specify the list of collectors to be enabled and override the resource detection by the charm.

**Comment by sudeephb** [comment]
Yes, after #79 is complete, the user should be able to manually install tools like ssacli, ipmi, etc (which would ideally be installed automatically). In that scenario, there are some steps the user has to take for installing these tools. For example, for ipmi, it's just installing the ipmitools and for ssacli, it's adding the relevant repository and installing the tool. We should mention these steps for each of the tools in the documentation.

**Comment by aieri** [comment]
I tend to disagree. If a resource that should be installed isn't (either because hardware detection fails, or for other reasons), the user should file a bug. I'd be in favor of closing this issue as 'won't fix'.

**Comment by jneo8** [comment]
> Yes, the user should be able to manually install tools like ssacli, ipmi, etc (which would ideally be installed automatically on machines that require them). In that scenario, there are some steps the user has to take for installing these tools. For example, for ipmi, it's just installing the ipmitool and for ssacli, it's adding the relevant repository and installing the tool. We should mention these steps for each of the tools in the documentation.

User shouldn't manually touch or install the resource, it's charm's job. I agree with @aieri if there are any case make tool install fail then it's a bug in the charm.

**Comment by jneo8** [comment]
More note to add, because we are using the strategy pattern. All the steps can be easily found under `hw_tools.py`. For example if I want to find the steps of install PercCLI. All the steps can be found [here](https://github.com/canonical/hardware-observer-operator/blob/d2f34e82a3634e17eeccc26ff548b64cf86870a8/src/hw_tools.py#L164)

We can have a document to mention this for developer if it's needed.

**Comment by Pjack** [comment]
Conclustion: We can reveal technical detail in document, but not teach user how to do it manually.  If any user submit issue, we can provide workaround solution within the issue.
